libcontainer/utils/utils_unix.go

```diff
@@ -31,3 +34,26 @@ func CloseExecFrom(minFd int) error {
 	}
 	return nil
 }
+
+// SaneTerminal sets the necessary tty_ioctl(4)s to ensure that a pty pair
+// created by us acts normally. In particular, a not-very-well-known default of
+// Linux unix98 ptys is that they have +onlcr by default. While this isn't a
+// problem for terminal emulators, because we relay data from the terminal we
+// also relay that funky line discipline.
+func SaneTerminal(terminal *os.File) error {
```

Comment:
I'm actually not sure (I think I wanted to make it so other people could use it but I'm not sure why that's necessary). Fixed.